proxy/src/http/project.rs

```diff
@@ -235,12 +241,40 @@ mod handler {
     }
 
     /// List all known projects.
-    pub async fn list<R>(ctx: http::Ctx<R>) -> Result<impl Reply, Rejection>
+    ///
+    /// If [`super::ListUser::user`] is given we only return projects that this user tracks.
+    pub async fn list<R>(
+        ctx: http::Ctx<R>,
+        super::ListQuery { user }: super::ListQuery,
+    ) -> Result<impl Reply, Rejection>
     where
         R: Send + Sync,
     {
         let ctx = ctx.read().await;
-        let projects = project::list_projects(&ctx.peer_api)?;
+        let all_projects = project::list_projects(&ctx.peer_api)?;
+
+        let projects = if let Some(user) = user {
```

Comment:
Usually finding large code blocks like this in handlers is a sign that the handler does too much, we want to treat them as light translations from the HTTP protocol into our core domain/implementations. This would be better suited to be in `project::list_projects`.